Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -715,19 +648,37 @@ namespace AzToolsFramework
                 const auto valueAtSubPath = GetDPE()->GetAdapter()->GetContents()[subPath];
 
                 // check if it's a PropertyHandler; if it is, just set it from the DOM directly
-                auto foundEntry = m_widgetToPropertyHandler.find(childWidget);
-                if (foundEntry != m_widgetToPropertyHandler.end())
+                auto foundEntry = m_widgetToPropertyHandlerInfo.find(childWidget);
+                if (foundEntry != m_widgetToPropertyHandlerInfo.end())
                 {
-                    foundEntry->second->SetValueFromDom(valueAtSubPath);
+                    auto handlerId = AZ::Interface<PropertyEditorToolsSystemInterface>::Get()->GetPropertyHandlerForNode(valueAtSubPath);
+
+                    // check if this patch has morphed the PropertyHandler into a different type
```

Comment:
Nice. Did you run into a bug here or just notice this when refactoring?

Reply:
+1 This is a really nice catch. For reference, this is a fully valid patch that normally would infrequently occur; changing a PropertyEditor from a Slider to a SpinBox or whatever isn't something we normally do, but the way patch generation works right now it's possible to add or remove row entries and have the generated patches be replaces instead. I have some thoughts on how to improve this going forward (by specifying a comparison key when comparing sets of nodes) but it's good to have this path working!

Reply:
@amzn-tmryan sure did! It turns out filtering is a really good stress test for patching code